### PR TITLE
Reworded IA topic and Added info about Silencing All alerts

### PR DESCRIPTION
**Pull request by catalinaadam**
Rewrote Integrated Alerting topic (PMM-8672) and added info about Silencing Alerts (PMM-9182)

### CODE REVIEW

**Review by rasika-chivate — CHANGES_REQUESTED** [review]

File: docs/using/alerting.md [hunk 1]
```diff
@@ -1,182 +1,339 @@
...
+                 by {{ $labels.service_name }} on {{ $labels.node_name }}.
+    ```
+## Test alert expressions
+If you want to create custom templates, you can test the MetricsQ expressions for your custom template in the **Explore** section of PMM. Here you can also query any PMM internal database.
+ 
+To test expressions for custom templates:
+1. On the left menu of PMM, choose **Explore > Metrics**.
+2. Enter your expression in the **Metrics** field and click **Run query**.
+ 
+For example, to validate that a MongoDB instance is down, shut down a member of a three-node replica set, then check that the expression **{service_type="mongodb"}**  returns **0** in **Explore > Metrics**.
```
Comment (fabio-silva) [inline docs/using/alerting.md:132]:
```suggestion
For example, to validate that a MongoDB instance is down, shut down a member of a three-node replica set, then check that the expression **{service_type="mongodb"}** returns **0** in **Explore > Metrics**.
```

File: docs/using/alerting.md [hunk 2]
```diff
@@ -1,182 +1,339 @@
...
-
-    ![!](../_images/PMM_Integrated_Alerting_Alert_Rule_Templates.jpg)
-
-1. Click <i class="uil uil-plus-square"></i> *Add*.
-
-1. Enter a template in the *Alert Rule Template* text box.
-
+1. Go to **Alerting > Integrated Alerting** and select the **Alert Rules** tab.
+2. Click **Add** on the right-hand side of the table.
+3. In the **Add Alert** rule dialog, choose the template on which you want to base the new alert rule. This automatically populates the **Threshold**, **Duration**, **Severity** fields with information from the template. You can change these values if you want to override the default specifications in the template.
```
Comment (fabio-silva) [inline docs/using/alerting.md:149]: `Threshold` and/or others is a parameter coming from the template defined by the user, so we shouldn't talk about it here. The fields that show up are the params defined beforehand (but Duration and Severity are indeed always there).

File: docs/using/alerting.md [hunk 3]
```diff
@@ -1,182 +1,339 @@
...
+Silencing an alert only stops notifications from being sent to your specified notification channels.
+ 
+Silenced alerts are still recorded under **Integrated Alerting > Alerts** so that you can review them later. A silenced alert is disabled until you reactivate the alert.
+ 
+ 
+To silence alerts:
+1. Click **Alerting** on the main menu.
+2. Choose **Integrated Alerting > Alerts** and choose if you want to disable an individual alert or all alerts:
+    - To silence an individual alert, scroll through the list of current alerts and click on the bell in the Actions column.
+    - To silence all Alerts, click Silence All at the top right side of the Alerts table.
```
Comment (fabio-silva) [inline docs/using/alerting.md:165]: We also have the option to "unsilence all", should we refer it here?

File: docs/using/alerting.md [hunk 4]
```diff
@@ -1,182 +1,339 @@
...
+ 
+## Alternative alerting options
+PMM Integrated Alerting was introduced as an alternative to Grafana Alerting and Prometheus’ Alertmanager.
+ 
+These third-party alerting tools continue to be available in PMM until Integrated Alerting moves from Technical Preview to General Availability.
+ 
+We recommend already using Integrated Alerting since it offers a simpler alerting setup and provides more robust alerting options.
+d Alerting is currently in [Technical Preview](../details/glossary.md#technical-preview). This means that this feature is still under development and is subject to change. Therefore, we recommend that early adopters use Integrated Alerting for testing purposes only.
```
Comment (fabio-silva) [inline docs/using/alerting.md:173]:
```suggestion
Alerting is currently in [Technical Preview](../details/glossary.md#technical-preview). This means that this feature is still under development and is subject to change. Therefore, we recommend that early adopters use Integrated Alerting for testing purposes only.
```

File: docs/using/alerting.md [hunk 5]
```diff
@@ -1,182 +1,339 @@
...
+ 
+### Set up a communication channel
+Set up a communication channel to configure how alerts will be delivered. The following notification channels are supported: email via your SMTP server, Slack, PagerDuty, and Webhook.
+ 
+To set up a communication channel:
+ 
+1. Open the new **Communication** tab under **Configuration > Settings**.
+2. Choose whether you want to configure the settings for **Email** or **Slack** sender channel. Webhook and PagerDury can be configured when adding the notification channel.
+    - For **Email**, fill in the details of your SMTP email server. Enforce TLS encryption if your SMTP server requires this, then click **Test** to confirm that you can connect to the SMTP server with the current settings.
+    - For **Slack**, specify the Slack webhook URL to use. Currently, you can only configure one Slack channel for each Slack URL.
```
Comment (fabio-silva) [inline docs/using/alerting.md:209]:
```suggestion
    - For **Slack**, specify the Slack Webhook URL to use. Currently, you can only configure one Slack channel for each Slack URL.
```

File: docs/using/alerting.md [hunk 6]
```diff
@@ -1,182 +1,339 @@
...
+ 
+To add a notification channel:
+ 
+1. Go to **Alerting > Integrated Alerting** and select the **Notification Channels** tab.
+2. Click **Add** and fill in the details for your required channel.
+3. Click **Add** to add the notification channel.
+ 
+## About alert templates
+Alerts templates are YAML files that provide the source framework for alert rules.
+Alert templates contain general template details and an alert expression defined in [MetricsQ](https://docs.victoriametrics.com/MetricsQL.html). This query language is backward compatible with Prometheus QL.
```
Comment (fabio-silva) [inline docs/using/alerting.md:231]:
```suggestion
Alert templates contain general template details and an alert expression defined in [MetricsQL](https://docs.victoriametrics.com/MetricsQL.html). This query language is backward compatible with Prometheus QL.
```

Comment (fabio-silva) [inline docs/using/alerting.md:231]:
```suggestion
Alert templates contain general template details and an alert expression defined in [MetricsQ](https://docs.victoriametrics.com/MetricsQL.html). This query language is backward compatible with PromQL.
```

File: docs/using/alerting.md [hunk 7]
```diff
@@ -1,182 +1,339 @@
...
+ 
+- **name** (required field): uniquely identifies template. Spaces and special characters are not allowed.
+- **version** (required): defines the template format version.
+- **summary** (required field): a template description.
+- **expr** (required field): a MetricsQL query string with parameter placeholders.
+- **params**: contains parameter definitions required for the query. Each parameter has a name, type, and summary. It also may have a unit, available range, and default value.
+    - **name** (required): the name of the parameter. Spaces and special characters not allowed.
+    - **summary** (required): a short description of what this parameter represents.
+    - **unit** (optional): PMM currently supports either s (seconds) or % (percentage).
+    - **type** (required): PMM currently supports the float type. string,  bool and another types will be available in a future release.
```
Comment (fabio-silva) [inline docs/using/alerting.md:252]:
```suggestion
    - **type** (required): PMM currently supports the *float* type. *string*, *bool* and another types will be available in a future release.
```

File: docs/using/alerting.md [hunk 8]
```diff
@@ -1,182 +1,339 @@
...
+    - **range** (optional): defines the boundaries for the value of a  float parameter
```
Comment (fabio-silva) [inline docs/using/alerting.md:253]:
```suggestion
    - **range** (optional): defines the boundaries for the value of a float parameter
```

File: docs/using/alerting.md [hunk 9]
```diff
@@ -1,182 +1,339 @@
...
+   - **value** (optional): default  parameter value. Value strings must not include any of these special characters: < > ! @ # $ % ^ & * ( ) _ / \ ' + - = (space)
```
Comment (fabio-silva) [inline docs/using/alerting.md:254]:
```suggestion
   - **value** (optional): default parameter value. Value strings must not include any of these special characters: < > ! @ # $ % ^ & * ( ) _ / \ ' + - = (space).
```

File: docs/using/alerting.md [hunk 10]
```diff
@@ -1,182 +1,339 @@
...
+- **for** (required): specifies the duration of time that the expression must be met before the alert will be fired
+- **severity** (required): specifies default alert severity level
+ - **labels** (optional): are additional labels to be added to generated alerts
```
Comment (fabio-silva) [inline docs/using/alerting.md:257]:
```suggestion
- **labels** (optional): are additional labels to be added to generated alerts.
```

File: docs/using/alerting.md [hunk 11]
```diff
@@ -1,182 +1,339 @@
...
+- **expr** (required field): a MetricsQL query string with parameter placeholders.
+- **params**: contains parameter definitions required for the query. Each parameter has a name, type, and summary. It also may have a unit, available range, and default value.
+    - **name** (required): the name of the parameter. Spaces and special characters not allowed.
+    - **summary** (required): a short description of what this parameter represents.
+    - **unit** (optional): PMM currently supports either s (seconds) or % (percentage).
+    - **type** (required): PMM currently supports the float type. string,  bool and another types will be available in a future release.
+    - **range** (optional): defines the boundaries for the value of a  float parameter
+   - **value** (optional): default  parameter value. Value strings must not include any of these special characters: < > ! @ # $ % ^ & * ( ) _ / \ ' + - = (space)
+- **for** (required): specifies the duration of time that the expression must be met before the alert will be fired
+- **severity** (required): specifies default alert severity level
```
Comment (fabio-silva) [inline docs/using/alerting.md:256]:
```suggestion
- **severity** (required): specifies default alert severity level.
```

File: docs/using/alerting.md [hunk 12]
```diff
@@ -1,182 +1,339 @@
...
+- **summary** (required field): a template description.
+- **expr** (required field): a MetricsQL query string with parameter placeholders.
+- **params**: contains parameter definitions required for the query. Each parameter has a name, type, and summary. It also may have a unit, available range, and default value.
+    - **name** (required): the name of the parameter. Spaces and special characters not allowed.
+    - **summary** (required): a short description of what this parameter represents.
+    - **unit** (optional): PMM currently supports either s (seconds) or % (percentage).
+    - **type** (required): PMM currently supports the float type. string,  bool and another types will be available in a future release.
+    - **range** (optional): defines the boundaries for the value of a  float parameter
+   - **value** (optional): default  parameter value. Value strings must not include any of these special characters: < > ! @ # $ % ^ & * ( ) _ / \ ' + - = (space)
+- **for** (required): specifies the duration of time that the expression must be met before the alert will be fired
```
Comment (fabio-silva) [inline docs/using/alerting.md:255]:
```suggestion
- **for** (required): specifies the duration of time that the expression must be met before the alert will be fired.
```

File: docs/using/alerting.md [hunk 13]
```diff
@@ -1,182 +1,339 @@
...
+        labels:
+              cultom_label: demo
+        annotations:
+              summary: MongoDB high memory usage ({{ $labels.service_name }})
+              description: |-
+                 {{ $value }}% of memory (more than [[ .threshold ]]%) is used
+                 by {{ $labels.service_name }} on {{ $labels.node_name }}.
     ```
+## Test alert expressions
+If you want to create custom templates, you can test the MetricsQ expressions for your custom template in the **Explore** section of PMM. Here you can also query any PMM internal database.
```
Comment (fabio-silva) [inline docs/using/alerting.md:292]:
```suggestion
If you want to create custom templates, you can test the MetricsQL expressions for your custom template in the **Explore** section of PMM. Here you can also query any PMM internal database.
```

File: docs/using/alerting.md [hunk 14]
```diff
@@ -1,182 +1,339 @@
...
+ 
+To test expressions for custom templates:
+1. On the left menu of PMM, choose **Explore > Metrics**.
+2. Enter your expression in the **Metrics** field and click **Run query**.
+ 
+For example, to validate that a MongoDB instance is down, shut down a member of a three-node replica set, then check that the expression **{service_type="mongodb"}**  returns **0** in **Explore > Metrics**.
```
Comment (fabio-silva) [inline docs/using/alerting.md:298]:
```suggestion
For example, to validate that a MongoDB instance is down, shut down a member of a three-node replica set, then check that the expression **{service_type="mongodb"}** returns **0** in **Explore > Metrics**.
```

File: docs/using/alerting.md [hunk 15]
```diff
@@ -1,182 +1,339 @@
...
+### With other alerting tools
+ 
+If you have existing YAML alert templates that you want to leverage in PMM Integrated Alerting:
+1. Go to **Alerting > Integrated Alerting > Alert Rule Templates** tab and click **Add** at the top right-hand side of the table.
+2. Click **Add** and upload a local .yaml file from your computer.
+ 
+## Add an Alert Rule
+1. Go to **Alerting > Integrated Alerting** and select the **Alert Rules** tab.
+2. Click **Add** on the right-hand side of the table.
+3. In the **Add Alert** rule dialog, choose the template on which you want to base the new alert rule. This automatically populates the **Threshold**, **Duration**, **Severity** fields with information from the template. You can change these values if you want to override the default specifications in the template.
```
Comment (fabio-silva) [inline docs/using/alerting.md:315]: Again, let's not talk about the threshold

File: docs/using/alerting.md [hunk 16]
```diff
@@ -1,182 +1,339 @@
...
+ 
+1. Go to **Alerting > Integrated Alerting** and select the **Notification Channels** tab.
+2. Click **Add** and fill in the details for your required channel.
+3. Click **Add** to add the notification channel.
+ 
+## About alert templates
+Alerts templates are YAML files that provide the source framework for alert rules.
+Alert templates contain general template details and an alert expression defined in [MetricsQ](https://docs.victoriametrics.com/MetricsQL.html). This query language is backward compatible with Prometheus QL.
+ 
+Starting with PMM 2.26, alert rules are no longer dependent on their source rule template after creation. This means that you can update or delete rule templates without impacting existing rules that are based on that template.
```
Comment (fabio-silva) [inline docs/using/alerting.md:67]:
```suggestion
Starting with PMM 2.26.0, alert rules are no longer dependent on their source rule template after creation. This means that you can update or delete rule templates without impacting existing rules that are based on that template.
```

File: docs/using/alerting.md [hunk 17]
```diff
@@ -1,182 +1,339 @@
...
+ 
+To test expressions for custom templates:
+1. On the left menu of PMM, choose **Explore > Metrics**.
+2. Enter your expression in the **Metrics** field and click **Run query**.
+ 
+For example, to validate that a MongoDB instance is down, shut down a member of a three-node replica set, then check that the expression **{service_type="mongodb"}**  returns **0** in **Explore > Metrics**.
+ 
+## Template compatibility
+### With previous PMM versions
+PMM 2.26 introduced major changes to the core structure of rule templates. As a result, alert rules and templates created in previous PMM versions are not compatible with PMM 2.26.0 and cannot be migrated to this new version.
```
[the remark this answers is not in the document]
Comment (fabio-silva) [reply docs/using/alerting.md:136]: Let's use 2.26.0 and 2.25.0 all over the place as well

**Review by fabio-silva — CHANGES_REQUESTED** [review]

**Review by catalinaadam — COMMENTED** [review]
Implemented suggested changes.

**Pull request merged**